buildenv/jenkins/jobs/infrastructure/copyrightCheck

```diff
@@ -60,28 +67,310 @@ timeout(time: 6, unit: 'HOURS') {
+                                                if (foundIBMCopyright >= 0) {
+                                                    // if we have an IBM copyright then this is an error
+                                                    println "We have a IBM Copyright - so this is an error"
+                                                    errorStr = "IBM Copyright should NOT be used in this file as it is a user config file"
+                                                    FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                                }
+                                            } else {
+                                                // The file is not a user configuration file - should have IBM portions
+                                                println "The file is not a user configuration file"
+                                                if (foundIBMPortionsCopyright <= 0) {
+                                                    errorStr = "IBM Portions Copyright should be used in this file"
+                                                    FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                                    checkCopyrightDate = false
+                                                }
+                                                // Check to see if the IBM Portions copyright is at top of file
+                                                if (foundIBMPortionsCopyright >= 0) {
+                                                    // We have an IBM copyright, so check is location
+                                                    if (foundIBMPortionsCopyright > 3 ) {
+                                                        // The Portions Copyright should be at top of the file
+                                                        errorStr = "Portions Copyright is not at top of the file"
+                                                        FAIL = addError(BAD_FILES, "${file}", errorStr)
+                                                    }
+                                                } // foundIBMPortionsCopyright
+                                            } // check user config
+                                        } // foundCPE
+                                    } // foundGPLv2
+                                } else {
+                                    // We have found some nonIBM copyright
+                                    if (foundGPLv2 >= 0) {
+                                        // We have found GPLv2
+                                        if (foundCPE < 0) {
+                                            // We have not found Classpath Exception
+                                            if (inJDK) {
+                                                // The file is on the JDK binary
+                                                errorStr = "WE CANNOT MODIFY THIS FILE - does not contain Classpath Exception"
```

Comment:
Not sure it matters if the file is modified or not. Files with a missing CPE just shouldn't be included in the JDK.

Reply:
Well this checker runs on a Pull Request so the file will have been modified in some way. 
I cannot remove the file from the Pull Request so all I can do is report that it has no CPE.
If the file is already part of the JDK source, all we can do is report that it has no CPE when it should. The person modifying the file can then determine what to do about it. ie get Oracle etc to fix etc. Will leave as is unless you think there is a better error message to be used.